niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/SignaturePermissionTester.java

```diff
@@ -971,7 +1009,11 @@ public void writeToParcel(Parcel parcel, int i) {
                 }));
 
         mPermissionTasks.put(permission.MANAGE_ACTIVITY_STACKS,
-                new PermissionTest(false, () -> {
+                new PermissionTest(false,Build.VERSION_CODES.P,
+                        Build.VERSION_CODES.Q,() -> {
```

Comment:
Is the platform still checking for this permission? Even if a permission has been deprecated, we should still leave the tests for that permission in place until the platform is no longer using the permission.

Reply:
I think it's depend on the policy. I felt the project try to maintain backward compatibility so I kept it.

Reply:
Sorry, what I mean is, as long as the platform still has a reachable check for the permission, we should not set a maxSdkVersion for the permission test even if it has been deprecated. I saw this line in `ActivityTaskManagerService` that allows access to the APIs even with the deprecated permission:

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/services/core/java/com/android/server/wm/ActivityTaskManagerService.java;l=3207;drc=998ef9a536e43339d14ae932e7116fe2a29faad9

You're right, this project does need to maintain backwards compatibility from P, so we do need to maintain all permission tests for previous releases.

Reply:
okay, will remove the max sdk version.  but I felt it's a debatable topic.

if once permission marked as deprecated, we can not predict when it will be removed.
and I assume after it removed,  we will set the max sdk version according to the platform manifest...

Reply:
I agree we cannot predict when it will be removed (although most deprecated items from the platform are typically not removed for several releases), but if the platform still has a code path dependent on that check, the tester would still need to validate that path since an app could attempt to exercise that code path. Once it is removed from the manifest, or there no longer exists a path that checks for the permission, then we can set a max SDK version for the permission.

The change looks good to me; thanks!